common/static/common/js/discussion/utils.js

```diff
@@ -151,61 +153,59 @@
         };
 
         DiscussionUtil.discussionAlert = function(header, body) {
-            var alertDiv, alertTrigger;
+            var $alertDiv, $alertTrigger;
             if ($("#discussion-alert").length === 0) {
-                alertDiv = $(
-                    "<div class='modal' role='alertdialog' id='discussion-alert' " +
-                    "aria-describedby='discussion-alert-message'/>"
+                $alertDiv = $(
+                    edx.HtmlUtils.joinHtml(
+                        edx.HtmlUtils.HTML("<div class='modal' role='alertdialog' id='discussion-alert' "),
+                        edx.HtmlUtils.HTML("aria-describedby='discussion-alert-message'/>")
+                    ).toString()
                 ).css("display", "none");
-                alertDiv.html(
-                    "<div class='inner-wrapper discussion-alert-wrapper'>" +
-                    "   <button class='close-modal dismiss' title='" + gettext("Close") + "'>" +
-                    "       <span class='icon fa fa-times' aria-hidden='true'></span>" +
-                    "   </button>" +
-                    "   <header><h2/><hr/></header>" +
-                    "   <p id='discussion-alert-message'/><hr/>" +
-                    "   <button class='dismiss'>" + gettext("OK") + "</button>" +
-                    "</div>"
+                edx.HtmlUtils.append(
+                    $alertDiv,
+                    edx.HtmlUtils.joinHtml(
+                        edx.HtmlUtils.HTML("<div class='inner-wrapper discussion-alert-wrapper'>"),
+                            edx.HtmlUtils.interpolateHtml(
+                                edx.HtmlUtils.HTML("<button class='close-modal dismiss' title='{title}'>"),
+                                {title: gettext("Close")}
+                            ),
+                                edx.HtmlUtils.HTML("<span class='icon fa fa-times' aria-hidden='true'></span>"),
+                            edx.HtmlUtils.HTML("</button>"),
+                            edx.HtmlUtils.HTML("<header><h2/><hr/></header>"),
+                            edx.HtmlUtils.HTML("<p id='discussion-alert-message'/><hr/>"),
+                            edx.HtmlUtils.HTML("<button class='dismiss'>"),
+                                gettext("OK"),
+                            edx.HtmlUtils.HTML("</button>"),
+                        edx.HtmlUtils.HTML("</div>")
```

Comment:
This is accurate.  Ideally, larger HTML blocks would move to Underscore.js files.  I like that this was tabbed so that it is mostly readable when concentrating on the HTML itself.  Another style that has been used can be seen here:
https://github.com/edx/edx-platform/blob/7d8f2e95785ac94f4db4253d68b67def46e948f8/common/lib/xmodule/xmodule/js/src/video/09_video_caption.js#L84-L118

@bjacobel @andy-armstrong: When you get a chance, maybe you can weigh in on the style we'd like to lean towards for this type of code.  We may also want to update the documentation with our preferences.

Thanks.